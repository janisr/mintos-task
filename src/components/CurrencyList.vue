<template>
  <ul>
    <currency-item
      v-for="currency in currencies"
      :key="currency"
      :currency="currency"
      :type="type"
      @currency-selected="$emit('currency-selected', $event)"
    ></currency-item>
  </ul>
</template>

<script lang="ts">
import { Options, Vue } from "vue-class-component";
import CurrencyItem from "./CurrencyItem.vue";
import { CurrencyType, TListType } from "./currency.types";

@Options({
  props: { currencies: Array, type: String },
  emits: ["currency-selected"],
  components: { CurrencyItem }
})
export default class CurrencyList extends Vue {
  public currencies!: CurrencyType[];
  public type!: TListType;
}
</script>

<style lang="scss" scoped>
ul {
  list-style-type: none;
  display: flex;

  margin: 4px;
  padding: 0;
  flex-wrap: wrap;
}
</style>
