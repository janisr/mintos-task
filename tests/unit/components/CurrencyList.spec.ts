import { CurrencyType } from "@/components/currency.types";
import { mount } from "@vue/test-utils";
import CurrencyList from "../../../src/components/CurrencyList.vue";

describe("CurrencyList.vue", () => {
  it("should compute selected currencies", () => {
    const currencies: CurrencyType[] = [
      { label: "EUR", selected: true },
      { label: "SEK", selected: false },
    ];

    const wrapper = mount(CurrencyList, {
      data: () => {
        return { currencies };
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
