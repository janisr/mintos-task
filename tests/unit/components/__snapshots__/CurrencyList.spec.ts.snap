// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyList.vue should compute selected currencies 1`] = `
<ul>
  
  <li
    class="selected undefined"
  >
    <label>
      EUR
    </label>
    <!--v-if-->
  </li>
  <li
    class="undefined"
  >
    <label>
      SEK
    </label>
    <!--v-if-->
  </li>
  
</ul>
`;
